django/contrib/gis/db/backends/postgis/pgraster.py

```diff
@@ -129,7 +126,7 @@ def to_pgraster(rast):
 
         # Set the nodata flag
         if band.nodata_value is not None:
-            pixeltype += 64
+            pixeltype |= BANDTYPE_FLAG_HASNODATA
```

Comment:
`pixeltype |= BANDTYPE_FLAG_HASNODATA` looks unrelated to the fix. Can you split your PR into two commits, first with refactoring (unrelated to fix) and second with a fix and tests.

I will be easier for me to review this ticket.